src/tests/pintest.c

```diff
@@ -64,7 +65,9 @@ static int ask_and_verify_pin(struct sc_pkcs15_object *pin_obj)
 		return 0;
 	}
 
-	sprintf(prompt, "Please enter PIN code [%.*s]: ", (int) sizeof pin_obj->label, pin_obj->label);
+	snprintf(prompt, sizeof(prompt), "Please enter PIN code [%.*s]: ",
```

Comment:
Why don't you simply change the above code to `char prompt[(sizeof pin_obj->label) + 30];`

Reply:
Here it would make sense and it would make it simpler.  Thanks for the suggestion.

In the worst case, potentially malicious card/driver can have more space to display in prompt something we would not like (`\r` and overwriting our actual prompt).

Reply:
Appending bad stuff at the end is irrelevant if only the first bytes get copied over. However, generally speaking, you are right. The solution would be to validate all the card's inputs, which is kind of labor intensive with the given code base...